src/foundry/client/pixi/core/interaction/render-flags.d.ts

```diff
@@ -0,0 +1,106 @@
+export {};
+
+declare global {
+  /** @privateRemarks Values are marked as optional here based on use, foundry docs incomplete */
+  type RenderFlag = {
+    /** Activating this flag also sets these flags to true */
+    propagate?: string[];
+    /** Activating this flag resets these flags to false */
+    reset?: string[];
+  };
+
+  /**
+   * A data structure for tracking a set of boolean status flags.
+   * This is a restricted set which can only accept flag values which are pre-defined.
+   */
+  class RenderFlags extends Set {
+    /**
+     * @param flags  - An object which defines the flags which are supported for tracking
+     * @param config - Optional configuration
+     */
+    constructor(
+      flags: Record<string, RenderFlag>,
+      config?: {
+        /** The object which owns this RenderFlags instance */
+        object?: RenderFlagObject;
+
+        /**
+         * The update priority when these render flags are applied.
+         * Valid options are OBJECTS or PERCEPTION.
+         * @defaultValue `PIXI.UPDATE_PRIORITY.OBJECTS`
+         */
+        priority?: "OBJECT" | "PERCEPTION";
+      },
+    );
+
+    /**
+     * {@inheritDoc}
+     * @returns The flags which were previously set that have been cleared.
+     */
+    clear(): Record<string, boolean>;
+
+    /**
+     * Allow for handling one single flag at a time.
+     * This function returns whether the flag needs to be handled and removes it from the pending set.
+     * @param flag - No comment
+     */
+    handle(flag: string): boolean;
+
+    /**
+     * Activate certain flags, also toggling propagation and reset behaviors
+     * @param changes - No comment
+     */
+    set(changes: Record<string, boolean>): void;
+  }
+
+  /**
+   * Add RenderFlags functionality to some other object.
+   * This mixin standardizes the interface for such functionality.
+   * @remarks Actually a function `RenderFlagsMixin(Base)`
+   * @param Base - The base class being mixed
+   * @returns The mixed class definition
+   */
+  class RenderFlagObject {
+    constructor(...args: any[]);
+
+    /**
+     * Configure the render flags used for this class.
+     * @defaultValue
+     * ```ts
+     * {
+     *   object: this,
+     *   priority: this.constructor.RENDER_FLAG_PRIORITY
+     * }
+     * ```
+     */
+    static RENDER_FLAGS: Record<string, RenderFlag>;
```

Comment:
Either the `defaultValue` or the typing is wrong. `this` will be `RenderFlagObject` (or a subclass) which is technically assignable to `RenderFlag` but that's only because both `propagate` and `reset` are optional. But then `RENDER_FLAG_PRIORITY` is just straight up a string (to be precise `"OBJECTS" | "PERCEPTION"`) which is definitely wrong as it really shouldn't be assignable to `RenderFlag`.

Actually in general this class may misuse `this` because `this` should probably refer to the class after being mixed in but I'm not sure it will because it's being done through the type level intersection trick. I'll take a closer look later.